libcudacxx/include/cuda/std/__mdspan/mdspan.h

```diff
@@ -108,28 +110,45 @@ class mdspan
       return (__self.rank() > 0)
           && __MDSPAN_FOLD_OR((__self.__mapping_ref().extents().template __extent<_Idxs>() == index_type(0)));
     }
+
+    template <class... _SizeTypes>
+    _CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI static constexpr bool
+    __check_index(_Extents const& exts, _SizeTypes... __indices)
+    {
+      index_type __zero = 0; // avoid warning:186 (ICC/NVCC) pointless comparison of unsigned integer with zero
+      array<bool, sizeof...(_SizeTypes)> __res{((is_signed_v<index_type> && static_cast<index_type>(__indices) < __zero)
+                                                || static_cast<index_type>(__indices) >= exts.extent(_Idxs))...};
+      _LIBCUDACXX_UNUSED_VAR(__zero * 2);
```

Comment:
I believe we should directly check against `npos` rather than < 0

Reply:
`npos` is for strings and maps to `-1`, while for `mdspan` we need to check against `dynamic_extent` which maps to the maximum value https://en.cppreference.com/w/cpp/container/span/dynamic_extent